### PR TITLE
doc: Update Istio GSG for Istio 0.7.0 and sidecar HTTP proxy

### PR DESCRIPTION
**Pull request by rlenglet**
Update the GSG to run Cilium with Istio using Docker images built from https://github.com/cilium/istio/tree/inject-cilium-filters and the Envoy binary built from https://github.com/cilium/cilium/pull/3035.

Signed-off-by: Romain Lenglet <romain@covalent.io>

### CODE REVIEW

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
Successful build: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/2054/

File: examples/kubernetes-istio/cilium-kube-inject.sed
Comment (tgraf) [inline examples/kubernetes-istio/cilium-kube-inject.sed:5]: We'll want to share a subdirectory in the future so we don't have to expose the entire state directory with each sidecar.

File: examples/kubernetes/cilium-sidecar.yaml
Comment (tgraf) [inline examples/kubernetes/cilium-sidecar.yaml]: In future, we could make this `--sidecar-http-proxy=$(SIDECAR_MODE)" so we can use a single DaemonSet and only provide a different configmap for the sidecar mode.

**Review by tgraf — APPROVED** [review]

File: examples/kubernetes/cilium-sidecar.yaml
Comment (rlenglet) [reply examples/kubernetes/cilium-sidecar.yaml]: The big plan is to make this setting per-endpoint at some point, so I didn't make this configuration polished.

**Comment by ianvernon** [comment]
So stoked to see this! Would it be a good idea to see if we can automate testing of this in our nightly CI builds? This of course would be handled separately as a follow-up PR.

File: examples/kubernetes-istio/cilium-kube-inject.sed
Comment (raybejjani) [reply examples/kubernetes-istio/cilium-kube-inject.sed:5]: Is it possible to mount the .sock directly? I've seen it done with docker but I don't know if volumeMounts/volumes allows it.

File: examples/kubernetes-istio/bookinfo-details-v1.yaml
Comment (raybejjani) [inline examples/kubernetes-istio/bookinfo-details-v1.yaml:52]: Out of curiosity, are the patterns auto-anchored or might this match anything that ends with `/details/[0-9]*`?

**Review by raybejjani — APPROVED** [review]

File: examples/kubernetes-istio/bookinfo-details-v1.yaml
Comment (rlenglet) [reply examples/kubernetes-istio/bookinfo-details-v1.yaml:52]: With Envoy they’re supposed to be.

File: examples/kubernetes-istio/cilium-kube-inject.sed
Comment (rlenglet) [reply examples/kubernetes-istio/cilium-kube-inject.sed:5]: Maybe. I’ll look into that. I have 2 sockets to mount. I went with the simplest solution for now.

File: examples/kubernetes/cilium-sidecar.yaml
Comment (aanm) [inline examples/kubernetes/cilium-sidecar.yaml]: Can we wait until this PR is merged https://github.com/cilium/cilium/pull/3125? that way we can move this file to the `plugins` / `addons` directory in `examples/kubernetes`

**Review by aanm — DISMISSED** [review]

File: examples/kubernetes/cilium-sidecar.yaml
Comment (rlenglet) [reply examples/kubernetes/cilium-sidecar.yaml]: Sure, np.

**Comment by aanm** [comment]
@rlenglet please rebase against master and put this in the `addons` directory

**Comment by rlenglet** [comment]
@aanm I addressed your review comments. My YAML files couldn't be moved into the `addons` directory, because those are different Cilium DaemonSet definitions. I had to refactor the `Makefile` to support generating multiple DaemonSets for each version of K8s.

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
To see what this Makefile does, I'd recommend looking at the internal database:
```
make -n -p | grep -v '^#'
```

**Review by aanm — DISMISSED** [review]
@rlenglet Can you remove the `examples/cilium-sidecar.yaml` and update the documentation with something like

```
.. tabs::
  .. group-tab:: K8s 1.7

    .. parsed-literal::

      $ wget \ |SCM_WEB|\/examples/kubernetes/1.7/cilium.yaml
      $ vim cilium.yaml
      [adjust the etcd address]

  .. group-tab:: K8s 1.8

    .. parsed-literal::

      $ wget \ |SCM_WEB|\/examples/kubernetes/1.8/cilium.yaml
      $ vim cilium.yaml
      [adjust the etcd address]

  .. group-tab:: K8s 1.9

    .. parsed-literal::

      $ wget \ |SCM_WEB|\/examples/kubernetes/1.9/cilium.yaml
      $ vim cilium.yaml
      [adjust the etcd address]

  .. group-tab:: K8s 1.10

    .. parsed-literal::

      $ wget \ |SCM_WEB|\/examples/kubernetes/1.10/cilium.yaml
      $ vim cilium.yaml
      [adjust the etcd address]

  .. group-tab:: K8s 1.11

    .. parsed-literal::

      $ wget \ |SCM_WEB|\/examples/kubernetes/1.11/cilium.yaml
      $ vim cilium.yaml
      [adjust the etcd address]
```

File: Documentation/gettingstarted/istio.rst
Comment (manalibhutiyani) [inline Documentation/gettingstarted/istio.rst]: Nit: `minkube` --> `minikube`

File: examples/kubernetes-istio/cilium-kube-inject.sed
Comment (manalibhutiyani) [inline examples/kubernetes-istio/cilium-kube-inject.sed:1]: Nit: I think `mount` is the correct word and not `inject` a directory?

Comment (rlenglet) [reply examples/kubernetes-istio/cilium-kube-inject.sed:1]: I prefer to use the "inject" term because this is in the context of running the `istioctl kube-inject` commands, which modifies the deployment specs to _inject_ new containers, labels, etc. So "inject" seems like the proper term.

File: Documentation/gettingstarted/istio.rst
Comment (rlenglet) [reply Documentation/gettingstarted/istio.rst]: Fixed.

**Comment by rlenglet** [comment]
test-me-please

**Comment by eloycoto** [comment]
Backport https://github.com/cilium/cilium/pull/3696

**Pull request merged**
